packages/codegen-ui-react/lib/forms/form-renderer-helper/model-values.ts

```diff
@@ -87,39 +95,110 @@ export function getDisplayValueScalar(fieldName: string, model: string, key: str
       ),
       undefined,
       [
-        factory.createCallExpression(
-          factory.createPropertyAccessExpression(
-            factory.createIdentifier(getRecordsName(model)),
-            factory.createIdentifier('find'),
-          ),
-          undefined,
-          [
-            factory.createArrowFunction(
-              undefined,
+        dataApi === 'GraphQL' && formActionType === 'update'
```

Comment:
why specifically for update forms?

Reply:
Create forms doesn't have this issue as the selected record will always be from the autocomplete options, so I didn't want to mess with how create forms are functioning now.

Reply:
Saw the issue with create forms and updated with the fix.